awx/main/wsrelay.py

```diff
@@ -310,19 +310,30 @@ async def run(self):
         for k, v in settings.LISTENER_DATABASES.get('default', {}).get('OPTIONS', {}).items():
             database_conf['OPTIONS'][k] = v
 
-        async_conn = await psycopg.AsyncConnection.connect(
-            dbname=database_conf['NAME'],
-            host=database_conf['HOST'],
-            user=database_conf['USER'],
-            password=database_conf['PASSWORD'],
-            port=database_conf['PORT'],
-            **database_conf.get("OPTIONS", {}),
-        )
+        try:
+            async_conn = await psycopg.AsyncConnection.connect(
+                dbname=database_conf['NAME'],
+                host=database_conf['HOST'],
+                user=database_conf['USER'],
+                password=database_conf['PASSWORD'],
+                port=database_conf['PORT'],
+                **database_conf.get("OPTIONS", {}),
+            )
+        except Exception as e:
+            logger.error(e)
+            return
+
         await async_conn.set_autocommit(True)
         event_loop.create_task(self.on_ws_heartbeat(async_conn))
 
         # Establishes a websocket connection to /websocket/relay on all API servers
         while True:
+            try:
+                await async_conn.execute("SELECT 1")
```

Comment:
This should be a separate async task that runs periodically, not in this while True loop.